src/action/timeline.rs

```diff
@@ -188,52 +206,273 @@ impl super::Response for ChunkResponse {
 mod tests {
 
     use super::*;
+    use std::fs::{hard_link, create_dir, write, set_permissions, Permissions};
+    use std::ffi::OsStr;
+    use std::path::Path;
+    use std::os::unix::{ffi::OsStrExt, fs::{symlink, PermissionsExt}};
     use tempfile::tempdir;
     use crate::action::Request;
-    use std::os::unix::ffi::OsStrExt;
     use crate::gzchunked::GzChunkedDecoder;
 
+    fn handle_for_path<S>(session: &mut S, path: &Path) -> session::Result<()>
+    where
+        S: Session,
+    {
+        let path_bytes = Vec::from(path.as_os_str().as_bytes());
+        let args_proto = TimelineArgs { root: Some(path_bytes) };
+        let request = Request::from_proto(args_proto).unwrap();
+        handle(session, request)
+    }
```

Comment:
This does not seem correct. If you do this, it feels like you are trying to test too much in tests that are not suited for this. If your are testing your handler, then why go through all this conversion to raw bytes and then conversion back from raw bytes?